services/streammanager/wunderkind/wunderkindmanager.go

```diff
@@ -0,0 +1,110 @@
+package wunderkind
+
+import (
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/cmd/rudder-cli/config"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+const (
+	INVOCATION_TYPE         = "RequestResponse"
+	WUNDERKIND_REGION       = "WUNDERKIND_REGION"
+	WUNDERKIND_IAM_ROLE_ARN = "WUNDERKIND_IAM_ROLE_ARN"
+	WUNDERKIND_EXTERNAL_ID  = "WUNDERKIND_EXTERNAL_ID"
+	WUNDERKIND_LAMBDA       = "WUNDERKIND_LAMBDA"
+)
+
+var (
+	jsonfast = jsoniter.ConfigCompatibleWithStandardLibrary
+)
+
+type inputData struct {
+	Payload string `json:"payload"`
+}
+
+type WunderkindProducer struct {
+	client LambdaClient
+	logger logger.Logger
+}
+
+type LambdaClient interface {
+	Invoke(input *lambda.InvokeInput) (*lambda.InvokeOutput, error)
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*WunderkindProducer, error) {
```

Comment:
Do we need destination ans Opts?, we are not using it? 
```suggestion
func NewProducer() (*WunderkindProducer, error) {
```

Reply:
No need, remove the extra parameters from function